### DISCUSSION TITLE
Compiling S-function using given matlab script

### DISCUSSION THREAD

**Discussion by MasterNomi** [JSBSim-Team/jsbsim#983] [Q&A] [answered]
Hi! I am a new to using mex files and using cmake. I have installed it using window installer. And when I run the s-function compiling script on matlab, i get a long list of LNK errors for .obj in jsbsim.lib file. I am using matlab 2022b and VS 2015. If this issue is because of VS 2015, is there a way around because installing VS 2019 or later is not feasible for me at the moment.

**Comment by agodemar** [comment]
Hello @MasterNomi I usually create the S-Function by generating the VS projects via CMake and setting the variable BUILD_MATLAB_SFUNCTION to true.

@bcoconni and @seanmcleod  I think we should remove this MATLAB script and direct users to get their local Mef-File via CMake

**Reply by bcoconni** [reply]
> @bcoconni and @seanmcleod I think we should remove this MATLAB script and direct users to get their local Mef-File via CMake

I am no Matlab user but I do not see why we should do that ? Wouldn't it be better to mention the CMake process in the [S-Function docs](https://github.com/JSBSim-Team/jsbsim/blob/release/v1.2/matlab/README.md) rather than deleting the script ?

**Reply by bcoconni** [reply]
I mean, we could update the docs to mention CMake as the preferred way to build the S-Function and keep the script file for people who cannot use CMake ?

**Comment by bcoconni** [comment, marked as answer]
Hi @MasterNomi,

Indeed the `JSBSim.lib` file that you get from the Windows installer has been compiled with VS 2019 so it's likely that the binary is not compatible with earlier versions of MSVC. This can be worked around by using CMake as @agodemar suggested. For that, you will need to download the [source code of JSBSim v1.2](https://github.com/JSBSim-Team/jsbsim/archive/refs/tags/v1.2.0.zip) which is the current stable branch. Once the files are unzipped, you can execute the following commands in a terminal
```pwsh
cd $JSBSIM_SOURCE_ROOT_DIRECTORY
mkdir build
cd build
cmake -DBUILD_MATLAB_SFUNCTION=ON ..
```
However if you cannot or do not want to use CMake, you can build `JSBSim.lib` from scratch with VS 2015. For that, you can open the MSVC project `JSBSim.vcxproj` (it is included in the source code zip file) in Visual Studio 2015 and compile `JSBSim.lib`. From there, you should be able to run the s-function compiling script on Matlab.
